Makefile

```diff
@@ -1,37 +1,51 @@
 help:
-	@echo "install_poetry: installs poetry on ubuntu"
+	@echo "install_rps: installs dependencies, builds proto, then installs the package"
 	@echo "clean: clean up grpc-generated code (by deletion)"
 	@echo "build_proto: generate code from the .proto files in protocols/proto-remote-processor"
-	@echo "install_rps: installs dependencies, builds proto, then installs the package"
-	@echo "common_build: install main dependencies"
-	@echo "server_build: install the package, but not the dependencies"
-
-install_poetry:
-	apt update
-	DEBIAN_FRONTEND=noninteractive apt -y install --no-install-recommends python3-poetry gcc python3-dev
-	apt clean
-	-rm -rf /var/lib/apt/lists/*
-	poetry config virtualenvs.path /rps/poetry_venv
 
 clean:
 	-rm -rd proto_remote_processor
 
 build_proto:
-	poetry install --no-root --only build
 	poetry run python -m grpc_tools.protoc -I protocols/ --python_out=. --pyi_out=. --grpc_python_out=. protocols/proto-remote-processor/*.proto
 
 install_rps:
 	poetry install --no-root
 	make build_proto
 	poetry install --only-root
 
+### Docker steps sorted in the same order as the Dockerfile
+### Not for general use so undocumented
+
+aryn_user:
+	groupadd --gid 1000 aryn
+	useradd --uid 1000 --gid 1000 --home-dir /aryn --password=y --no-create-home aryn
+	chown -R aryn:aryn /aryn
+
+install_poetry:
+	touch /var/lib/apt/.cache_var_lib_apt # make it possible to find the cache directory for buildx builds
+	touch /var/cache/apt/.cache_var_cache_apt
+	apt update
+	DEBIAN_FRONTEND=noninteractive apt -y install --no-install-recommends python3-poetry gcc python3-dev
+
 common_build:
+	test "$(POETRY_CACHE_DIR)" = /tmp/poetry_cache # catch a bug where putting ARG too early in Dockerfile doesn't get the env var
```

Comment:
does it have to be specifically /tmp/poetry_cache? or would like a -z $POETRY_CACHE_DIR or something suffice?

Reply:
Everything needs to be consistent so it seemed best to check that it was exactly what was expected.  I see little benefit in supporting people trying to override the POETRY_CACHE_DIR during builds.   FWIW, you can't override it as an env var, you have to do: --build-arg=POETRY_CACHE_DIR=/a/b/c/d/e (which with the current code will fail)